platform/mds/mds-web/src/main/java/org/motechproject/mds/web/service/impl/InstanceServiceImpl.java

```diff
@@ -952,25 +953,32 @@ private Object parseValueForDisplay(Object value, MetadataDto relatedFieldMetada
             //Using ZonedDateTime for retrieving timezone
             ZonedDateTime zonedDateTime = ZonedDateTime.of((LocalDateTime) parsedValue, ZoneOffset.systemDefault());
             parsedValue = java.time.format.DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm xxxx").format(zonedDateTime);
-        } else if (relatedFieldMetadata != null) {
-            parsedValue = removeCircularRelations(parsedValue, relatedFieldMetadata.getValue());
+        } else if (relatedClassMetadata != null) {
+            // We do not want to return the whole chain of relationships for UI display, but just the first level.
+            // Fetching whole relationship tree may cause trouble when serializing
+            parsedValue = breakDeepRelationChainForDisplay(parsedValue, relatedClassMetadata.getValue());
         }
 
         return parsedValue;
     }
 
-    private Object removeCircularRelations(Object object, String relatedField) {
-        // we must also handle a field that is a collection
-        // because of this we handle regular fields as single objects collection here
-        Collection objectsCollection = (object instanceof Collection) ? (Collection) object : Arrays.asList(object);
+    private Object breakDeepRelationChainForDisplay(Object value, String relatedClassName) {
```

Comment:
Ok, here's what we should do. We should either try using @Transactional(propagation=NEVER), but that would have to go on the methods that use this, since this is not a public method. Eventually we can try manually checking the transaction manager here, and throw an exception if a transaction exists. I'm wary of leaving it like this to be honest.